metron-platform/metron-parsing/metron-parsers-common/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -81,4 +81,11 @@
    */
   boolean validate(T message);
 
+  /**
+   * Provides a hook to override the default charset parsers use to read data.
+   * @return Charset to use for for reading
+   */
+  default Charset getReadCharset() {
+    return StandardCharsets.UTF_8;
```

Comment:
Should this potentially be a global configuration that defaults to `UTF_8`?, which is then overridden by parser specific charset?

I.e. if a user has all (or almost all) of their parsers in another charset, right now they'd have to set it for each and every parser. I'm not sure if this an actual or theoretical issue to be honest.

Reply:
I made a compromise on not handling this also with a global option. As we had talked above about needing to change read charsets likely being an edge case, it seemed reasonable to add the capability to the parsers individually.

Reply:
Sounds good. I'd honestly just forgotten that conversation existed at this point and missed it when I too quickly looked.